.github/workflows/dockerhub.yml

```diff
@@ -1,228 +1,341 @@
+    needs: determine_docker_org_and_tag
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+      - name: Get Docker Image Metadata
+        shell: bash
+        id: docker-image-metadata
+        run: |
+          echo build_date=$(date -u +”%Y-%m-%dT%H:%M:%SZ”) >> $GITHUB_OUTPUT
+          echo vcs_ref=$(git rev-parse --short HEAD) >> $GITHUB_OUTPUT
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+        uses: docker/setup-buildx-action@v3
       - name: Login to DockerHub
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build
-        uses: docker/build-push-action@v3
+      - name: Build and Push Docker Image
+        uses: docker/build-push-action@v5
         with:
           context: ./telematic_system/telematic_apps/grafana
-          file: ./telematic_system/telematic_apps/grafana/Dockerfile
           push: true
-          tags: usdotfhwastoldev/telematic_grafana:${{ github.ref_name }} 
+          tags: usdotfhwastoldev/telematic_grafana:${{ github.ref_name }}
+          build-args: |
+            BUILD_DATE=${{ steps.docker-image-metadata.outputs.build_date }}
+            VCS_REF=${{ steps.docker-image-metadata.outputs.vcs_ref }}
 
   dockerhub-telematic-local-apache2:
+    needs: determine_docker_org_and_tag
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+      - name: Get Docker Image Metadata
+        shell: bash
+        id: docker-image-metadata
+        run: |
+          echo build_date=$(date -u +”%Y-%m-%dT%H:%M:%SZ”) >> $GITHUB_OUTPUT
+          echo vcs_ref=$(git rev-parse --short HEAD) >> $GITHUB_OUTPUT
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+        uses: docker/setup-buildx-action@v3
       - name: Login to DockerHub
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build
-        uses: docker/build-push-action@v3
+      - name: Build and Push Docker Image
+        uses: docker/build-push-action@v5
         with:
           context: ./telematic_system/telematic_apps/apache2
           file: ./telematic_system/telematic_apps/apache2/local.Dockerfile
           push: true
-          tags: usdotfhwastoldev/telematic_local_apache2:${{ github.ref_name }} 
-  
+          tags: usdotfhwastoldev/telematic_local_apache2:${{ github.ref_name }}
+          build-args: |
+            BUILD_DATE=${{ steps.docker-image-metadata.outputs.build_date }}
+            VCS_REF=${{ steps.docker-image-metadata.outputs.vcs_ref }}
+
   dockerhub-telematic-local-messaging:
```

Comment:
This image is not required and can be removed from the yml

Reply:
updated